source/extensions/dynamic_modules/abi.h

```diff
@@ -15,10 +15,19 @@
 
 #ifdef __cplusplus
 #include <cstddef>
+#include <cstdint>
 
 extern "C" {
 #else
 #include <stddef.h>
+#include <stdint.h>
+#endif
+
+#ifdef ENVOY_DYNAMIC_MODULE_GO_SDK
+// Use the pointer-sized integer to represent a raw pointer in C for simplicity in Go bindings.
```

Comment:
note: otherwise, unnecessary CGo runtime check prevents us from implementing a low-level exchange of pointers between Go and Envoy